users/views.py

```diff
@@ -116,11 +106,54 @@ def get(self: "ActivateUser", request: HttpRequest, uidb64: str, token: str) ->
             return redirect("home-page")
 
 
-class ProfileSettingsView(TemplateView):
-    # Work on this view is in progress
+class ProfileSettingsView(LoginRequiredMixin, UpdateView):
+    model = Profile
     template_name = "users/profile_settings.html"
+    form_class = UserProfileForm
+    success_url = reverse_lazy("profile")
+    login_url = "login"
+
+    def get_object(self: "ProfileSettingsView") -> Profile:
+        return Profile.objects.get(user=self.request.user)
 
+    def get(self: "ProfileSettingsView") -> HttpResponse:
+        self.object = self.get_object()
+        user = self.object.user
 
-class AccountSettingsView(TemplateView):
-    # Work on this view is in progress
+        user_form = UserForm(instance=user)
```

Comment:
Please review above sourcery suggestions if they make sense to reduce complexity.

Reply:
Done